### DISCUSSION TITLE
Support for Energy metering for ZigBee Plug Neo NAS-WR01

### DISCUSSION THREAD

**Discussion by kamiKAC** [arendst/Tasmota#13776] [Ideas]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
I have working Sonoff ZB Bridge with Tasmota. Few days ago I've bought Neo NAS-WR01plug. I can switch it on and off but energy meter is not functional.
Looks like it is supported in development branch of zigbee2mqtt:
https://github.com/Koenkk/zigbee-herdsman-converters/issues/3191

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Neo NAS-WR01 with  Sonoff SB Bridge
- [x] Tasmota binary firmware version number used: 10.0.0.3(zbbridge) Build Date & Time 2021-11-19T22:18:22
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: OTA
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
00:14:14.237 CMD: Backlog Template; Module; GPIO 255
00:14:14.238 SRC: WebConsole from 192.168.2.9
00:14:14.239 CMD: Grp 0, Cmnd 'BACKLOG', Idx 1, Len 26, Data 'Template; Module; GPIO 255'
00:14:14.295 SRC: Backlog
00:14:14.297 CMD: Grp 0, Cmnd 'TEMPLATE', Idx 1, Len 0, Data ''
00:14:14.301 MQT: stat/tasmota_AA05E2/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}
00:14:14.508 SRC: Backlog
00:14:14.510 CMD: Grp 0, Cmnd 'MODULE', Idx 1, Len 0, Data ''
00:14:14.514 MQT: stat/tasmota_AA05E2/RESULT = {"Module":{"75":"Sonoff ZbBridge"}}
00:14:14.758 SRC: Backlog
00:14:14.759 CMD: Grp 0, Cmnd 'GPIO', Idx 1, Len 3, Data '255'
00:14:14.766 MQT: stat/tasmota_AA05E2/RESULT = {"GPIO0":{"320":"Led_i1"},"GPIO1":{"3552":"Zigbee Tx"},"GPIO2":{"0":"None"},"GPIO3":{"3584":"Zigbee Rx"},"GPIO4":{"5312":"Zigbee Rst1"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"640":"I2C SDA"},"GPIO13":{"576":"LedLink_i"},"GPIO14":{"608":"I2C SCL"},"GPIO15":{"0":"None"},"GPIO16":{"32":"Button1"},"GPIO17":{"0":"None"}}
```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
00:15:25.618 CMD: Status 0
00:15:25.619 SRC: WebConsole from 192.168.2.9
00:15:25.621 CMD: Grp 0, Cmnd 'STATUS', Idx 1, Len 1, Data '0'
00:15:25.625 MQT: stat/tasmota_AA05E2/STATUS = {"Status":{"Module":75,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_AA05E2","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
00:15:25.632 MQT: stat/tasmota_AA05E2/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/tasmota-zbbridge.bin","RestartReason":"Software/System restart","Uptime":"2T23:35:08","StartupUTC":"2021-11-19T23:40:17","Sleep":50,"CfgHolder":4617,"BootCount":254,"BCResetTime":"2021-04-07T18:10:44","SaveCount":332,"SaveAddress":"FD000"}}
00:15:25.637 MQT: stat/tasmota_AA05E2/STATUS2 = {"StatusFWR":{"Version":"10.0.0.3(zbbridge)","BuildDateTime":"2021-11-19T22:18:22","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":160,"Hardware":"ESP8266EX","CR":"364/699"}}
00:15:25.643 MQT: stat/tasmota_AA05E2/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":4,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["iot",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A00000000000000","00000080","00086080","00004000"]}}
00:15:25.652 MQT: stat/tasmota_AA05E2/STATUS4 = {"StatusMEM":{"ProgramSize":670,"Free":1120,"Heap":25,"ProgramFlashSize":2048,"FlashSize":2048,"FlashChipId":"1540A1","FlashFrequency":40,"FlashMode":3,"Features":["00000809","0F1007C6","04400001","00000003","00000000","00000000","00020080","00200000","00000000"],"Drivers":"1,2,4,7,9,10,12,20,23,38,41,50","Sensors":"1"}}
00:15:25.658 MQT: stat/tasmota_AA05E2/STATUS5 = {"StatusNET":{"Hostname":"tasmota-AA05E2-1506","IPAddress":"192.168.2.65","Gateway":"192.168.2.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.2.1","DNSServer2":"0.0.0.0","Mac":"84:CC:A8:AA:05:E2","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
00:15:25.663 MQT: stat/tasmota_AA05E2/STATUS6 = {"StatusMQT":{"MqttHost":"piHAB.lan","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_AA05E2","MqttUser":"DVES_USER","MqttCount":5,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
00:15:25.670 MQT: stat/tasmota_AA05E2/STATUS7 = {"StatusTIM":{"UTC":"2021-11-22T23:15:25","Local":"2021-11-23T00:15:25","StartDST":"2021-03-28T02:00:00","EndDST":"2021-10-31T03:00:00","Timezone":"+01:00","Sunrise":"08:11","Sunset":"17:01"}}
00:15:25.676 MQT: stat/tasmota_AA05E2/STATUS10 = {"StatusSNS":{"Time":"2021-11-23T00:15:25"}}
00:15:25.684 MQT: stat/tasmota_AA05E2/STATUS11 = {"StatusSTS":{"Time":"2021-11-23T00:15:25","Uptime":"2T23:35:08","UptimeSec":257708,"Vcc":3.530,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":5,"Wifi":{"AP":1,"SSId":"iot","BSSId":"4E:5E:0C:24:9E:9B","Channel":11,"Mode":"11n","RSSI":64,"Signal":-68,"LinkCount":5,"Downtime":"0T00:00:17"}}}
00:15:29.851 MQT: tele/tasmota_AA05E2/STATE = {"Time":"2021-11-23T00:15:29","Uptime":"2T23:35:12","UptimeSec":257712,"Vcc":3.540,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":5,"Wifi":{"AP":1,"SSId":"iot","BSSId":"4E:5E:0C:24:9E:9B","Channel":11,"Mode":"11n","RSSI":62,"Signal":-69,"LinkCount":5,"Downtime":"0T00:00:17"}}


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Pair Neo NAS-WR01 with Sonoff ZB bridge with Tasmota

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
Energy measurements populated in Web interface.
Energy measurements populated via MQTT.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._
```lua
  Console output here:
00:26:47.753 CMD: ZbStatus3 4
00:26:47.755 SRC: WebConsole from 192.168.2.9
00:26:47.756 CMD: Grp 0, Cmnd 'ZBSTATUS', Idx 3, Len 1, Data '4'
00:26:47.764 MQT: stat/tasmota_AA05E2/RESULT = {"ZbStatus3":[{"Device":"0x0FC9","IEEEAddr":"0xA4C13875A6ADF221","ModelId":"TS011F","Manufacturer":"_TZ3000_w0qqde0g","Endpoints":[1],"Config":["O01"],"Power":1,"Reachable":true,"LastSeen":385,"LastSeenEpoch":1637623222,"LinkQuality":81}]}
```

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
It is a Tuya device. Tuya "enhances" Zigbee with own non standard extensions.
There is a chance, if the needed adoptions are not to big, to integrate.

**Reply by kamiKAC** [reply]
I'm not professional developer but I made small contribution to Tasmota. Unfortunately adding support for Zigbee device is much too difficult for me.
I have this Neo plug at the moment so if you would like to add support for it in Tasmota I am ready to help/test.
It's cheapest Zigbee plug with energy meter I could buy locally so having full support for it in Tamsota would be great.

**Reply by Jason2866** [reply]
It is a feature request. "Someone" with the skills for, needs to have a look at. May happen or not.
Your help will be needed in any case since you are the only one with the device.

**Comment by kamiKAC** [comment]
As (according to small note on the seller's page) I got version without power meter, Ive returned the plug and ordered another one which has power meter.
I was able to read voltage with following command:
ZbSend { "device":"0x7515", "endpoint":"0x01", "cluster":"0x0B04", "read":"0x0505" }

How do I set TelePeriod for Zigbee device?
P.S. Looks like this topic is about the same device https://github.com/arendst/Tasmota/discussions/13962

**Reply by kamiKAC** [reply]
So far I found those values:
Voltage reading (V):
ZbSend { "device":"0x7515", "endpoint":"0x01", "cluster":"0x0B04", "read":"0x0505" } 
Active power reading (W):
ZbSend { "device":"0x7515", "endpoint":"0x01", "cluster":"0x0B04", "read":"0x050B" }
Current reading (mA)
ZbSend { "device":"0x7515", "endpoint":"0x01", "cluster":"0x0B04", "read":"0x0508" }
Energy consumption
ZbSend { "device":"0x7515", "endpoint":"0x01", "cluster":"0x0702", "read":"0x0000" }

Rule which reads active power, voltage and current every minute
Rule on time#minute do ZbSend { "device":"0x7515", "endpoint":"0x01", "cluster":"0x0B04", "read":["0x050B","0x0505","0x0508"] } endon

I have no idea how to read this value more often (eg 10s).

Best Regards

**Reply by fredmaria** [reply]
How did you find the parameters for the zbsend command?

**Reply by kamiKAC** [reply]
List of clusters can be found here: https://developer.tuya.com/en/docs/iot-device-dev/tuya-zigbee-measuring-smart-plug-access-standard?id=K9ik6zvofpzqk
Unfortunately I can' t use this plug with Tasmota because it breaks functionality of Sonoff SNZB-04 when it starts to act as repeater. Messages from Sonoff are intercepted and resend as other messages originated from plug. I don't know if this is related to first issue but plug intermittently switches off.
As it was useless because of described issues I don't use this plug any more.

**Reply by fredmaria** [reply]
Thanks for the info.
I use this plug with sonoff zbbridge flashed with tasmota and module (75) together with mqtt.
And as far as i needed it, it works.
